### PR TITLE
Document hotplug network interfaces using live migration

### PR DESCRIPTION
**Pull request by ormergi**
Add documentation about the Hotplug Network Interfaces feature using live-migration to attache interfaces.

### CODE REVIEW

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:103]: Please consider reducing one `#` on each newly added heading.
```suggestion
## Migration based hotplug
```

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:127]: nit: Please consider using this method first.
I personally find it simpler.

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:104]:
```suggestion
In case your cluster doesn't use the Multus version as specified above, it's possible to hotplug an interface by migrating the VM.
```

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @ormergi.
Is there a difference when following the newly added procedure on a VM vs on a VMI?

File: docs/operations/hotplug_interfaces.md
Comment (maiqueb) [reply docs/operations/hotplug_interfaces.md:103]: I actually think this makes sense.

File: docs/operations/hotplug_interfaces.md
Comment (maiqueb) [reply docs/operations/hotplug_interfaces.md:104]: It's not only the multus version: if you have the correct multus version, but didn't deploy the dynamic networks controller, it still won't work.

Thus, I think the original suggestion was preferable - while a bit vague, it is specific enough to be correct.

File: docs/operations/hotplug_interfaces.md
Comment (maiqueb) [inline docs/operations/hotplug_interfaces.md:105]:
```suggestion
The actual attachment won't take place immediately, and the new interface will be available in the guest once the migration is completed.
```

I think we should be a bit more specific above, thus I've rephrased a bit the second premise.

File: docs/operations/hotplug_interfaces.md
Comment (maiqueb) [inline docs/operations/hotplug_interfaces.md:111]:
```suggestion
At this point the new interface is added to the spec but will not be attached to the running VM. 
```

File: docs/operations/hotplug_interfaces.md
Comment (maiqueb) [inline docs/operations/hotplug_interfaces.md:113]: Would it make sense to link to the migration docs instead ? You're repeating a documented process.

Comment (ormergi) [reply docs/operations/hotplug_interfaces.md:113]: I think that having all the commands right in front of you makes it easier to understand the flow (instead of jumping to another page).
How about keeping the migration-with-virtctl and add a link to the title?
If you think its also redundant I will remove it all (and keep the link only)

Comment (maiqueb) [reply docs/operations/hotplug_interfaces.md:113]: Your call. I don't mind either way.

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: I removed the manifest method and added a link to the full live migration doc.

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: Done

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: Done

**Review by maiqueb — APPROVED** [review]
Nice call adding the link to the requirements.

I would just consider doing what @orelmisan suggests in https://github.com/kubevirt/user-guide/pull/664/files#r1150088464 and drop one heading indentation.

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:113]: Please consider adding a sentence like "For more details, please see live migration" with a link, instead of using the heading as a link.

File: docs/operations/hotplug_interfaces.md
Comment (AlonaKaplan) [inline docs/operations/hotplug_interfaces.md:104]: The requirements include "Enabling network interface hotplug support", migration-based hotplug requires the feature gate, please add the explicit requirements that are not needed for migration-based hot-plug.

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:107]:
```suggestion
#### Hotplugging an interface
```

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: Done

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: I added reference to the migration docs.

Regarding the title I prefer to leave it imperative with a snippet for the migration command to make the flow clear (and not make the reader jump to other pages).

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: Done

**Comment by maiqueb** [comment]
It's fine by me; I'll leave the lgtm to @orelmisan if he's OK with it.

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [reply docs/operations/hotplug_interfaces.md:108]: 1. Could you please change the heading to `###`?
2. nit: Could you please remove the colon `:`?

File: docs/operations/hotplug_interfaces.md
Comment (orelmisan) [inline docs/operations/hotplug_interfaces.md:133]: Why was this line removed?

**Review by orelmisan — COMMENTED** [review]
Overall looks good.
There are only minor comments.

**Comment by ormergi** [comment]
> It's fine by me; I'll leave the lgtm to @orelmisan if he's OK with it.

@maiqueb if you have no objections and you are fine with those changes, feel free to place /lgtm regardless.

File: docs/operations/hotplug_interfaces.md
Comment (ormergi) [reply docs/operations/hotplug_interfaces.md]: My bad, fixed.

**Comment by ormergi** [comment]
@orelmisan I resolved your, please take another look.

**Review by orelmisan — APPROVED** [review]
Thank you for the changes @ormergi

**Pull request merged**
